src/JMS/Serializer/EventDispatcher/Subscriber/SymfonyValidatorSubscriber.php

```diff
@@ -21,14 +21,23 @@
 use JMS\Serializer\EventDispatcher\Event;
 use JMS\Serializer\EventDispatcher\EventSubscriberInterface;
 use JMS\Serializer\Exception\ValidationFailedException;
-use Symfony\Component\Validator\ValidatorInterface;
+use Symfony\Component\Validator\ValidatorInterface as LegacyValidatorInterface;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
 
 class SymfonyValidatorSubscriber implements EventSubscriberInterface
 {
     private $validator;
 
-    public function __construct(ValidatorInterface $validator)
+    public function __construct($validator)
     {
+        if (!$validator instanceof ValidatorInterface && !$validator instanceof LegacyValidatorInterface) {
+            throw new \InvalidArgumentException('Argument 1 of '.__METHOD__.' must be an instance of Symfony\Component\Validator\Validator\ValidatorInterface or Symfony\Component\Validator\ValidatorInterface.');
+        }
+
+        if ($validator instanceof LegacyValidatorInterface) {
```

Comment:
check for ``!$validator instanceof ValidatorInterface`` instead. Otherwise you will get the deprecation warning all the time (the class implements both interfaces in Symfony for BC reasons)